### ISSUE TITLE
Error During Install

### ISSUE THREAD

**Issue by nekoyokoshima** [sjarvela/kloudspeaker#100]
While http://<my domain>/kloudspeaker/backend/install/ i get the following error

Could not install

Error executing queries (CREATE TABLE `user` ( `id` int(11) NOT...): All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead

MySQL  5.7.10
CentOS Linux release 7.2.1511 (Core)
Nginx 1.8.0 (Using nginx.org repo)
php-fpm: 5.6.17-1 (Remi)

Steps to reproduce:
- Fresh LEMP server with mentioned versions
- Created DB Schema and user
- Used backend/example/example_configuration_mysql.php as base configuration
- Proceeded with install using /backend/install/

**Comment by nekoyokoshima** [comment]
Ok so, some updates
- The current version of Kloudspeaker will not install with MySQL 5.7. I'm using 5.6 and it installs fine.

The other thing was that once installed I was getting some NULL errors on the login page; which did not load anything but a message box. Was scratching my head for some hours, until I tried using Apache instead of Nginx. Same problem; however apache was more descriptive, churning out this

```
[Fri Jan 22 17:36:10.747850 2016] [:error] [pid 10099] [client 172.16.x.x:52561] KLOUDSPEAKER ERROR: PHP error #2, mysqli_connect(): Headers and client library minor version mismatch. Headers:50541 Library:50628 (/var/www/html/kloudspeaker/backend/db/mysql/MySQLIDatabase.class.php:128)\n{NOTE:Backtrace disabled as it may contain passwords. Enable 'debug' setting in configuration.php to see backtrace.}, referer: http://172.16.x.x/kloudspeaker/
```

That can be easily fixed by removing php-mysql an instead installing php-mysqlnd.

Now I'm at a point where Login screen is displayed and I'm trying to set folders {but failing, for now}

Am getting

```
Uncaught Cannot resolve permission, permissions not fetched
```

Todo:
- I'm gonna retry Nginx 1.8.0 now that i'm using the require php-mysqlnd module
- Will post details of fix to "Uncaught Cannot resolve permission, permissions not fetched"

**Comment by sjarvela** [comment]
The permissions error should be fixed in latest release. I'll investigate the MySQL 5.7 case, haven't seen that before.

**Comment by Aetf** [comment]
I have got the same error

```
Error executing queries (CREATE TABLE user ( id int(11) NOT...): All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead
```

after upgraded my system to mysql 5.7, and install kloudspeaker from scratch.

Using nginx 1.9.15, mysql 5.7, php 7.0

**Comment by madaello** [comment]
Same error here:
Error executing queries (CREATE TABLE `user` ( `id` int(11) NOT...): All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead

I'm using current standard packages of Ubuntu Xenial:
- Ubuntu LTS 16.04
- Apache 2.4.18
- Mysql 5.7.12
- PHP 7.0.4

**Comment by Tjr25** [comment]
i had the same problem and posted a fix in issue # 127 "issue creating admin user - 1st install #127". hope that helps you

**Comment by sjarvela** [comment]
So the root reason really is that there is NULL field in primary key list, and this is a clear error, I will fix this. For some reason some MySQL versions do accept NULL, as I haven't seen this before.

**Comment by sjarvela** [comment]
Fixed install script in 2.7.15
